src/rust/engine/process_executor/src/lib.rs

```diff
@@ -0,0 +1,159 @@
+use std::io::Error;
+use std::collections::HashMap;
+use std::process::Command;
+
+///
+/// A process to be executed.
+///
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub struct ExecuteProcessRequest {
+    argv: Vec<String>,
+    /// Ideally this would be a HashMap, but that isn't hashable, and we need to implement Hash to
```

Comment:
OK, so the sorting below says this comment can go or else should be modified heavily. This is all forward looking since it seems an ExecuteProcessRequest itself doesn't care about env ordering, but an action cache will. Perhaps just focus the new doc comment on the equality guaranty between 2 ExecuteProcessRequests with the same argv and env map.

Reply:
Switched to use a BTreeMap, which has Hashable.